lib/services/local/events.go

```diff
@@ -157,24 +161,27 @@ func (w *watcher) Error() error {
 	return nil
 }
 
-func (w *watcher) parseEvent(e backend.Event) (*services.Event, error) {
+func (w *watcher) parseEvent(e backend.Event) ([]services.Event, []error) {
+	if e.Type == backend.OpInit {
+		return []services.Event{{Type: e.Type}}, nil
+	}
+	events := []services.Event{}
+	errs := []error{}
 	for _, p := range w.parsers {
-		if e.Type == backend.OpInit {
-			return &services.Event{Type: e.Type}, nil
-		}
 		if p.match(e.Item.Key) {
 			resource, err := p.parse(e)
 			if err != nil {
-				return nil, trace.Wrap(err)
+				errs = append(errs, trace.Wrap(err))
+				continue
 			}
 			// if resource is nil, then it was well-formed but is being filtered out.
 			if resource == nil {
-				return nil, nil
+				continue
 			}
-			return &services.Event{Type: e.Type, Resource: resource}, nil
+			events = append(events, services.Event{Type: e.Type, Resource: resource})
 		}
 	}
-	return nil, trace.NotFound("no match found for %v %v", e.Type, string(e.Item.Key))
+	return events, errs
```

Comment:
why is this change required?

Reply:
Because a single key can now be matched by multiple parsers each of which produce a different resource/event. For instance, `/cluster_configuration/session_recording` is now matched by both `clusterConfigParser` and `sessionRecordingConfigParser`.